torch/_inductor/pattern_matcher.py

```diff
@@ -1621,6 +1621,8 @@ def is_mutation_op(node: torch.fx.Node) -> bool:
     elif node.op == "call_method":
         if _mutation_op_re.search(node.target):  # type: ignore[union-attr, arg-type]
             return True
+    if node.target is torch.ops.higher_order.auto_functionalized:
+        return False
```

Comment:
auto_functionalized nodes can have `out = blah` arguments. Those are not mutable (the auto_functionalized node is always functional), so we fix that here.

Reply:
This is unrelated?

Reply:
Fair enough, I'll split this out into its own PR